_books/ion-1-1/src/macros/system_macros.md

```diff
@@ -0,0 +1,398 @@
+
+Examples:
+```ion
+(:flatten [a, b, c] (d e f)) => a b c d e f
+(:flatten [[], null.list] foo::()) => [] null.list
+```
+
+The `flatten` macro can also be used to splice the content of one list or s-expression into another list or s-expression.
+```ion
+[1, 2, (:flatten [a, b]), 3, 4] => [1, 2, a, b, 3, 4]
+```
+
+#### `parse_ion`
+
+Ion documents may be embedded in other Ion documents using the `parse_ion` macro.
+
+```ion
+(macro parse_ion (data!) /* Not representable in TDL */)
+```
+
+The `parse_ion` macro constructs a stream of values by parsing a blob literal or string literal as a single, self-contained Ion document.
+
+Embedded text example:
+```ion
+(:parse_ion
+    '''
+    $ion_1_1
+    $ion_encoding::((symbol_table ["foo" "bar"]]))
+    $1 $2
+    '''
+)
+=> foo bar
+```
+
+
+Embedded binary example:
+```ion
+(:parse_ion {{ 4AEB6qNmb2+jYmFy }} )
+=> foo bar
+```
+
+> [!IMPORTANT]
+> Unlike most macros, this macro specifically requires _literals_. Macros are not allowed to contain recursive calls,
+> and composing an embedded document from multiple expressions would make it possible to implement recursion in the
+> macro system.
+> 
+> Allowing context to leak from the outer scope into the document being parsed would also enable recursion.
+>
+> Furthermore, `parse_ion` cannot be invoked from a macro body (i.e. in Template Definition Language (TDL)). 
+> (Why not?)
+
+
+### Value Constructors
+
+#### `annotate`
+
+```ion
+(macro annotate (ann* value) /* Not representable in TDL */)
+```
+
+Produces the `value` prefixed with the annotations `ann`s.
+Each `ann` must be a non-null, unannotated string or symbol.
+
+```ion
+(:annotate (: "a2") a1::true) => a2::a1::true
+```
+
+#### `make_string`
+
+```ion
+(macro make_string (content*) /* Not representable in TDL */)
+```
+
+Produces a non-null, unannotated string containing the concatenated content produced by the arguments.
+Nulls (of any type) are forbidden. Any annotations on the arguments are discarded.
+
+#### `make_symbol`
+
+```ion
+(macro make_symbol (content*) /* Not representable in TDL */)
+```
+
+Like `make_string` but produces a symbol.
+
+#### `make_blob`
+
+```ion
+(macro make_blob (lobs*) /* Not representable in TDL */)
+```
+
+Like `make_string` but accepts lobs and produces a blob.
+
+#### `make_list`
+
+```ion
+(macro make_list (sequences*) [ (.flatten sequences) ])
+```
+
+Produces a non-null, unannotated list by concatenating the _content_ of any number of non-null list or sexp inputs.
+
+```ion
+(:make_list)                  => []
+(:make_list (1 2))            => [1, 2]
+(:make_list (1 2) [3, 4])     => [1, 2, 3, 4]
+(:make_list ((1 2)) [[3, 4]]) => [(1 2), [3, 4]]
+```
+
+#### `make_sexp`
+
+```ion
+(macro make_sexp (sequences*) ( (.flatten sequences) ))
+```
+
+Like `make_list` but produces a sexp.
+This is the only way to produce an S-expression from a template: unlike lists, S-expressions in
+templates are not quasi-literals.
```

Comment:
I think this has been obviated by `.` syntax. Together with `flatten`, this also raises the question: do we need `make_list`/`make_sexp`?

Reply:
I'd be hesitant to remove them, but that's partly just because I haven't thought about it much to determine whether there is any use case for them that isn't solved in other ways.